README.md
# General
jd stuff


Server/history.txt
    1  raspi-config
    2  sudo raspi-config
    3  sudo apt-get dist-upgrade
    4  sudo nano /etc/dhcpcd.conf
    5  sudo reboot
    6  clear
    7  sudo apt-get update
    8  sudo apt-get install openvpn
    9  ls
   10  cd vpn
   11  ls
   12  cat vpn.conf
   13  ls
   14  cp vpn.conf /etc/openvpn
   15  sudo cp vpn.conf /etc/openvpn
   16  sudo service openvpn@vpn enable
   17  sudo service vpn@openvpn enable
   18  curl icanhazip.com
   19  sudo service openvpn start
   20  sudo service openvpn status
   21  sudo systemctl status vpn@openvpn
   22  sudo systemctl status openvpn
   23  sudo systemctl status openvpn@vpn
   24  curl icanhazip.com
   25  sudo service openvpn stop
   26  cd /etc/openvpn
   27  ls
   28  sudo systemctl enable openvpn
   29  sudo service openvpn start
   30  sudo systemctl status openvpn@vpn.service
   31  curl icanhazip.com
   32  sudo apt-get install git transmission transmission-daemon
   33  sudo service transmission-daemon stop
   34  sudo shutdown now
   35  python3 pi.py
   36  cd /media
   37  ls
   38  sudo mkdir TORRENTS
   39  cd ~
   40  lks
   41  ls
   42  sudo fdisk -l
   43  python3 pi.py
   44  sudo apt-get install exfat-utils
   45  python3 pi.py
   46  cd /media/TORRENTS
   47  ls
   48  cd ~
   49  sudo apt-get install samba
   50  ls
   51  sudo apt-get install samba-common-bin
   52  sudo nano /etc/samba/smb.conf
   53  sudo service samba restart
   54  sudo smbpasswd -a pi
   55  sudo service samba restart
   56  service smbd restart
   57  sudo service smbd restart
   58  cd ~/.config
   59  cat ~/.config
   60  sudo apt-get install clonezilla
   61  sudo apt-get install drbl
   62  cd /var/lib/
   63  ls
   64  cd transmission-daemon
   65  ls
   66  cat info
   67  cd info
   68  sudo cd info
   69  sudo cat info
   70  cd downloads
   71  ls
   72  sudo reboot
   73  python3 pi.py
   74  sudo service transmission-daemon stop
   75  sudo reboot
   76  python3 pi.py
   77  sudo cat /etc/transmission-daemon/settings.json
   78  ls
   79  mkdir gists
   80  sudo pip3 install python-gist
   81  sudo apt-get update
   82  sudo apt-get install pip3
   83  sudo apt-get install pip3-python
   84  sudo easy_install pip3
   85  sudo apt-get easy_install
   86  sudo apt-get install easy_install
   87  mkdir temp
   88  ls
   89  cd temp
   90  sudo git clone https://github.com/jdowner/gist
   91  cd gist
   92  ls
   93  sudo python3 setup.py install
   94  sudo python setup.py install
   95  sudo pip
   96  sudo apt-get install python3-pip
   97  ls
   98  sudo python3 setup.py install
   99  sudo python3 setup.py test
  100  gist
  101  gist list
  102  cd /home/pi
  103  ls
  104  sudo nano .gist
  105  gist list
  106  gist edit ffc8ffec35c242f113a1ca1e86e43b53
  107  ls
  108  cd gists
  109  ls
  110  cd ~
  111  cd temp
  112  ls
  113  cd gist
  114  ls
  115  cd ~
  116  ls
  117  cd ~/.ssh
  118  ls
  119  ssh-keygen -t rsa -C "jacko1394@me.com"
  120  ls
  121  sudo cat id_rsa.pub
  122  cd ~
  123  ls
  124  gist edit ffc8ffec35c242f113a1ca1e86e43b53
  125  git config --global user.email "jacko1394@me.com"
  126  git config --global user.name "Jack Della"
  127  gist edit ffc8ffec35c242f113a1ca1e86e43b53
  128  python3 pi.py
  129  clear
  130  python3 pi.py
  131  clear
  132  python3 pi.py
  133  clear
  134  python3 pi.py
  135  clear
  136  python3 pi.py
  137  ls
  138  clear
  139  python3 pi.py
  140  clear
  141  python3 pi.py
  142  clear
  143  python3 pi.py
  144  clear
  145  python3 pi.py
  146  c
  147  lear
  148  clear
  149  python3 pi.py
  150  clear
  151  python3 pi.py
  152  clear
  153  python3 pi.py
  154  clear
  155  python3 pi.py
  156  clear
  157  python3 pi.py
  158  clear
  159  python3 pi.py
  160  clear
  161  python3 pi.py
  162  clear
  163  ls
  164  python3 pi.py
  165  cat .gist
  166  clear
  167  python3 pi.py
  168  git commit
  169  git commit -av
  170  git clone ffc8ffec35c242f113a1ca1e86e43b53
  171  git clone https:www.github.com/ffc8ffec35c242f113a1ca1e86e43b53
  172  git clone https://gist.github.com/Jacko1394/ffc8ffec35c242f113a1ca1e86e43b53
  173  ls
  174  cd ffc8ffec35c242f113a1ca1e86e43b53
  175  ls
  176  sudo nano link.txt
  177  cat link.txt
  178  git add .
  179  ls .
  180  cat .
  181  git commit -m "5:30pm"
  182  git push origin master
  183  git config --global credential.helper cache
  184  git config --global user.name "Jacko1394"
  185  git push origin master
  186  exit
  187  ls
  188  sudo git config credential.helper store
  189  cd ffc8ffec35c242f113a1ca1e86e43b53
  190  sudo git config credential.helper store
  191  git config credential.helper store
  192  sudo nano link.txt
  193  git commit -m "5:53pm"
  194  git add .
  195  git commit -m "5:53pm"
  196  git push origin master
  197  sudo apt-get update
  198  sudo apt-get dist-upgrade
  199  cd ..
  200  ls
  201  cd Ds
  202  cat Ds
  203  rm Ds
  204  rm -r gists
  205  rm -r temp
  206  clear
  207  sudo rm -r temp
  208  ls
  209  exit
  210  history
  211  sudo apt-get update
  212  clear
  213  python3 pi.py
  214  git clone https://gist.github.com/Jacko1394/ffc8ffec35c242f113a1ca1e86e43b53
  215  rm ffc8ffec35c242f113a1ca1e86e43b53
  216  rm -r ffc8ffec35c242f113a1ca1e86e43b53
  217  sudo rm -r ffc8ffec35c242f113a1ca1e86e43b53
  218  ls
  219  python3 pi.py
  220  l
  221  sls
  222  ls
  223  ls ffc8ffec35c242f113a1ca1e86e43b53
  224  python3 pi.py
  225  cd ~/ffc8ffec35c242f113a1ca1e86e43b53
  226  ls
  227  cd ..
  228  python3 pi.py
  229  cd ~/ffc8ffec35c242f113a1ca1e86e43b53
  230  ls
  231  cat link.txt
  232  cat .
  233  cd ..
  234  python3 pi.py
  235  git add /home/pi/ffc8ffec35c242f113a1ca1e86e43b53
  236  l
  237  ls
  238  cd ~/ffc8ffec35c242f113a1ca1e86e43b53
  239  ls
  240  pwd
  241  git add /home/pi/ffc8ffec35c242f113a1ca1e86e43b53
  242  cd ~
  243  git add /home/pi/ffc8ffec35c242f113a1ca1e86e43b53
  244  python3 pi.py
  245  git commit -m "getting there"

Server/pi.py
#!/usr/bin/python3
import os.path
import datetime
import time
import subprocess
import re
import json
import gist
import requests

# Configs:
transSettings = '/etc/transmission-daemon/settings.json'
gistDirectory = '/home/xxx/ffc8ffec35c242f113a1ca1e86e43b53'
pwd = 'birdie133'
drivePath = '/media/HD'
hdDisk = '/dev/sdb2'
torrentPath = drivePath + '/Downloads'

# Globals:
public = ''
local = ''
port = ''
output = ''

def pPrint(text):
	print('\033[95m' + str(text) + '\033[0m')
	global output
	output += text + '\n'

def main():

	initialize()
	runUpdates()
	# installSoftware()

	if not mountDrive():
		return False

	getPortForward()
	updateTransmissionSettings()
	checkIpAdresses()
	updateGist()
	startNgrok()

	return True

def test():

	#Open, load config:
	f1 = open('/home/xxx/test.txt', 'r+')
	f1.write('ok')
	f1.close()

def initialize():

	status = False
	
	pPrint('Initialising...')

	while not status:
		os.system('echo %s|sudo -S %s' % (pwd, 'systemctl stop openvpn@vpn.service'))
		os.system('echo %s|sudo -S %s' % (pwd, 'service transmission-daemon stop'))
		#reset network here:::
		time.sleep(5)
		status = verifyInternet()
		if status:
			break

	#stop vpn, wait 10, check internet 3 times, waiting 10 inbetween
	#reboot

def verifyInternet():

	try:
		result = requests.head('http://www.google.com/', timeout=8)

		if 200 <= result.status_code <= 226:
			pPrint('Internet success code: ' + str(result.status_code))
			return True
		else:
			pPrint('Error: ' + result)
			return False

	except:
		pPrint('Error: Network not working!')
		return False

def runUpdates():

	pPrint('Updating packages (apt-get)...')

	#Updates: -y is assume Yes:
	os.system('echo %s|sudo -S %s' % (pwd, 'apt-get -y update'))
	os.system('echo %s|sudo -S %s' % (pwd, 'apt-get -y full-upgrade'))

def installSoftware():

	pPrint('Installing software...')

	#Installs:
	pPrint('Installing git...')
	os.system('echo %s|sudo -S %s' % (pwd, 'apt-get install -y git'))

	pPrint('Installing transmission...')
	os.system('echo %s|sudo -S %s' % (pwd, 'apt-get install -y git'))

def mountDrive():

	result = False

	pPrint('Mounting hard drive...')

	if os.path.ismount(drivePath):
		pPrint('Drive is mounted: ' + drivePath)
		result = True
	else:
		if os.system('echo %s|sudo -S %s' % (pwd, 'mount ' + hdDisk + ' ' + drivePath)):
			result = False
		else:
			pPrint('Successfully mounted drive: ' + drivePath)
			result = True
	
	if not result:
		pPrint('Error mounting drive :(\nMight not be plugged in?)')
		os.system('echo %s|sudo -S %s' % (pwd, 'systemctl stop openvpn@vpn.service'))
		os.system('echo %s|sudo -S %s' % (pwd, 'service transmission-daemon stop'))
		result = False

	return result

def getPortForward():

	global port

	pPrint('Getting port forward...')
	os.system('echo %s|sudo -S %s' % (pwd, 'systemctl restart openvpn@vpn.service'))
	time.sleep(5)

	try:
		port = subprocess.check_output('head -n 100 /dev/urandom | sha256sum | tr -d " -"', shell=True)
		port = port.decode('UTF-8').rstrip()
		port = subprocess.check_output('curl -m 8 "http://209.222.18.222:2000/?client_id=%s"' % (port), shell=True, close_fds=True, stdin=None, stderr=None)
		port = port.decode('UTF-8').rstrip()
	except:
		return False

	port = re.findall(r'\d+', port)[0]
	pPrint('PORT: ' + port)
	return True

def updateTransmissionSettings():

	global port

	pPrint('Updating transmission configuration...')

	#Restart the transmission-daemon service, and wait:
	os.system('echo %s|sudo -S %s' % (pwd, 'service transmission-daemon stop'))
	os.system('echo %s|sudo -S %s' % (pwd, 'service transmission-daemon start'))
	time.sleep(5)

	#Change permissions of settings.json so python can modify it:
	os.system('echo %s|sudo -S %s' % (pwd, 'chown xxx -R %s' % (transSettings)))

	#Open, load config:
	f1 = open(transSettings, 'r+')
	data = json.load(f1)

	#Change values:
	data['rpc-port'] = int(port)
	data['rpc-whitelist-enabled'] = False
	data['rpc-authentication-required'] = True
	data['rpc-username'] = 'jacko1394'
	data['rpc-password'] = 'pelican420'
	data['incomplete-dir'] = torrentPath
	data['download-dir'] = torrentPath

	#Save 
	f1.seek(0)
	f1.write(json.dumps(data))
	f1.truncate()
	f1.close()

	#Reset settings.json permissions, reload configuration:
	os.system('echo %s|sudo -S %s' % (pwd, 'chown debian-transmission -R %s' % (transSettings)))
	os.system('echo %s|sudo -S %s' % (pwd, 'service transmission-daemon reload'))
	return 0

def checkIpAdresses():

	global public
	global local

	pPrint('Checking IP addresses...')

	public = subprocess.check_output('curl -m 8 ipinfo.io/ip', shell=True, close_fds=True, stdin=None, stderr=None)
	public = public.decode('UTF-8').rstrip()
	pPrint('Public IP: ' + public)

	local = subprocess.check_output('/sbin/ifconfig | grep 192.168', shell=True, cwd='/home/xxx')
	local = local.decode('UTF-8').rstrip()
	local = re.search('inet (.*) netmask', local)
	local = str(local.group(1)).rstrip()
	pPrint('Local IP: ' + local)

	return True

def updateGist():

	global public
	global local
	global port

	pPrint('Updating Gist link.txt...')

	# Loads data:
	f1 = open(gistDirectory + '/link.txt', 'r+')
	f2 = open('/home/xxx/cronn.log', 'r')
	
	date = subprocess.check_output('date', shell=True)
	date = date.decode('UTF-8').rstrip()
	uname = subprocess.check_output('uname -a', shell=True)
	uname = uname.decode('UTF-8').rstrip()
	drives = subprocess.check_output('df -h | grep /dev/sd', shell=True)
	drives = drives.decode('UTF-8').rstrip()

	# Writes data:
	f1.seek(0)
	f1.write('SYSTEM:\n' + uname + '\n\nTIME:\n')
	f1.write(datetime.datetime.now().strftime('%-d/%-m/%Y %A %-I:%M%p') + '\n')
	f1.write(date + '\n\n')
	f1.write('DRIVES:\nFilesystem      Size  Used Avail Use% Mounted on\n')
	f1.write(drives + '\n\n')
	f1.write('URL:\n' + public + ':' + port + '\n')
	f1.write(local + ':' + port + '\n\nOUTPUT:\n' + output) # + '\nCRON LOG:\n')
	# f1.write(f2.read()) # cron log

	# Finalise:
	f1.truncate()
	f1.close()

	# Save changes to gist repo:
	r1 = subprocess.check_output('git add -A && git commit -m "xxx"', cwd=gistDirectory, shell=True)
	r1 = subprocess.check_output('git push origin master', cwd=gistDirectory, shell=True)
	pPrint('Updated Gist repo.')

	return 0

def startNgrok():

	pPrint('Starting ngrok connection...')
	r1 = subprocess.check_output('/home/xxx/ngrok tcp 22', shell=True, stdin=None, stderr=None, close_fds=True)

if __name__ == '__main__':
	main()
